Source/AwsCommonRuntimeKit/http/HTTPStream.swift

```diff
@@ -43,14 +35,22 @@ public class HTTPStream {
         return Int(status)
     }
 
-    // TODO: make it thread safe
     /// Activates the client stream.
     public func activate() throws {
+        lock.lock()
+        defer {
+            lock.unlock()
+        }
+        guard !activated else {
+            return
+        }
+
```

Comment:
no need to lock until required critical section arrives
```suggestion
        guard !activated else {
            return
        }
        lock.lock()
        defer {
            lock.unlock()
        }

```

Reply:
Thread sanitizer might think that this is a race condition. Activated is not supposed to be called multiple times, so this should not be in the hot path.
reference: https://github.com/awslabs/aws-crt-swift/pull/132#discussion_r1072972998

Reply:
Fixed, in #142